src/sparseml/exporters/transforms/kv_cache/cache_keys_and_values.py

```diff
@@ -29,6 +29,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+ALLOWED_NODES_BEFORE_SOFTMAX = ["Cast"]
```

Comment:
extensible, nice

Reply:
Yes, spoiler alert, for BLOOM you'd have to add a Reshape into this set ;)